opensquirrel/decomposer/aba_decomposer.py

```diff
@@ -0,0 +1,102 @@
+from __future__ import annotations
+
+import math
+from collections.abc import Callable, Iterable
+
+from opensquirrel.common import ATOL
+from opensquirrel.decomposer.general_decomposer import Decomposer
+from opensquirrel.ir import BlochSphereRotation, Float, Gate
+from opensquirrel.utils.identity_filter import filter_out_identities
+
+
+class ABADecomposer(Decomposer):
+    def __init__(self, ra: Callable[..., BlochSphereRotation], rb: Callable[..., BlochSphereRotation]):
+        self.ra = ra
+        self.rb = rb
+
+    @staticmethod
+    def get_decomposition_angles(alpha: float, axis: Iterable[float]) -> tuple[float, float, float]:
+        """
+        Gives the angles used in the A-B-A decomposition of the Bloch sphere rotation
+        characterized by a rotation around `axis` of angle `alpha`.
+
+        Parameters:
+            alpha: angle of the Bloch sphere rotation
+            axis: _normalized_ axis of the Bloch sphere rotation
+
+        Returns:
+            A triple (theta1, theta2, theta3) corresponding to the decomposition of the
+            arbitrary Bloch sphere rotation into U = Ra(theta3) Rb(theta2) Ra(theta1)
+
+        """
+        nx, ny, nz = axis
+
+        assert abs(nx**2 + ny**2 + nz**2 - 1) < ATOL, "Axis needs to be normalized"
+
+        assert -math.pi + ATOL < alpha <= math.pi + ATOL, "Angle needs to be normalized"
+
+        if abs(alpha - math.pi) < ATOL:
+            # alpha == pi, math.tan(alpha / 2) is not defined.
+
+            p: float
+            if abs(nz) < ATOL:
```

Comment:
This is still hardcoded for ny and nz angles

Reply:
In order to make it dynamic (i.e gate_list function) I feel like a similar integration to PR #98 would be required.

Reply:
Hi @juanboschero, this was my original PR. If you read the description, it says that it simply introduces the `ABADecomposer`, but then you have to generalize the `get_decomposition_angles` method.